phoenix-core/src/main/java/org/apache/phoenix/hbase/index/IndexQosRpcControllerFactory.java

```diff
@@ -60,27 +61,22 @@ public PayloadCarryingRpcController newController(List<CellScannable> cellIterab
 
     private class IndexQosRpcController extends DelegatingPayloadCarryingRpcController {
 
-        private Configuration conf;
         private int priority;
 
         public IndexQosRpcController(PayloadCarryingRpcController delegate, Configuration conf) {
             super(delegate);
-            this.conf = conf;
             this.priority = PhoenixIndexRpcSchedulerFactory.getMinPriority(conf);
         }
-
         @Override
         public void setPriority(final TableName tn) {
             // if its an index table, then we override to the index priority
-            if (isIndexTable(tn)) {
+            if (!tn.isSystemTable() &&  !PhoenixDatabaseMetaData.SYSTEM_TABLE_NAMES.contains(tn.getNameAsString())) {
```

Comment:
Actually, is it only SYSTEM.CATALOG that comes through here? How about if we just check for that with Schema.isMetaTable(tn.getName())? It's possible that we'd add an index to a system table down-the-road, so we may not want to be too general with this check.